purchase_quotation_products/models/product_product.py

```diff
@@ -60,40 +60,42 @@ def fields_view_get(self, view_id=None, view_type='form',
             res['arch'] = etree.tostring(doc)
         return res
 
-    @api.one
-    def _get_qty_purchase(self):
-        self.qty_purchase = 0
+    @api.multi
+    def _compute_qty_purchase(self):
         purchase_order_id = self._context.get('active_id', False)
-        if purchase_order_id:
-            lines = self.env['purchase.order.line'].search([
-                ('order_id', '=', purchase_order_id),
-                ('product_id', '=', self.id)])
-            self.qty_purchase = \
-                sum([self.env['product.uom']._compute_qty_obj(
-                    line.product_uom,
-                    line.product_qty,
-                    self.uom_po_id) for line in lines])
+        for rec in self:
```

Comment:
I think we can improve _compute_qty_purchase method

1. In this loop we are doing two writes, one to set to 0, and another to update the value. We can make only one and use update() instead.
2. context is been read only one time but we are making an if condition asking if this one is set or not, every time. We do not need to this question inside the for, only need to ask one time before the for
3. did not know why but some searchs do strange things inside compute method, I recommend to use a browse for the purchase.order and a filtered by product instead.

Example something like
```python
        if not purchase_order_id:
            self.update({'qty_purchase': 0}
            return

        for rec in self:
            lines = self.env['purchase.order'].browse(
                purchase_order_id).order_lines.filtered(
                    lambda x: x.product_id == rec.id))
            value = sum([self.env['product.uom']._compute_quantity(
                    line.product_qty,
                    line.product_uom,
                    rec.uom_po_id) for line in lines])
        rec.update(value)
 ```